sdk/testing/src/main/java/io/opentelemetry/sdk/testing/assertj/TraceAssert.java

```diff
@@ -66,6 +66,42 @@ public TraceAssert hasSpansSatisfyingExactly(
     return this;
   }
 
+  /**
+   * Asserts that the trace under assertion has the same number of spans as provided {@code
+   * assertions} and verifies that there is a combination of spans that satisfies specified {@link
+   * SpanDataAssert} {@code assertions} in the given order. This is a variation of {@link
+   * #hasSpansSatisfyingExactly(Consumer...)} where order does not matter.
+   */
+  @SafeVarargs
+  @SuppressWarnings("varargs")
+  public final TraceAssert hasSpansSatisfyingExactlyInAnyOrder(
+      Consumer<SpanDataAssert>... assertions) {
+    return hasSpansSatisfyingExactlyInAnyOrder(Arrays.asList(assertions));
+  }
+
+  /**
+   * Asserts that the trace under assertion has the same number of spans as provided {@code
+   * assertions} and verifies that there is a combination of spans that satisfies specified {@link
+   * SpanDataAssert} {@code assertions} in the given order. This is a variation of {@link
+   * #hasSpansSatisfyingExactly(Iterable)} where order does not matter.
+   */
+  public TraceAssert hasSpansSatisfyingExactlyInAnyOrder(
+      Iterable<? extends Consumer<SpanDataAssert>> assertions) {
+    @SuppressWarnings("unchecked")
+    Consumer<SpanData>[] spanDataAsserts =
```

Comment:
Ended up using this pattern a lot so have this helper, can you use it here?

https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/testing/src/main/java/io/opentelemetry/sdk/testing/assertj/AssertUtil.java#L25

https://github.com/open-telemetry/opentelemetry-java/blob/f464817013aca1c77203a1f7ced178fb54bd9f9f/sdk/testing/src/main/java/io/opentelemetry/sdk/testing/assertj/DoubleGaugeAssert.java#L42